### ISSUE TITLE
message with R 4.1+ anonymous function definition style

### ISSUE THREAD

**Issue by JWiley** [jimhester/lintr#878]
`R`  introduced a new shorthand way to define a function, useful for anonymous functions.

```r
lapply(1:5, \(x) x + 1)
``` 

In `lintr` 2.0.1 on `R` 4.1.1 this gets flagged with the message "Place a space before left parenthesis, except in a function call."

I am agnostic on whether you want to support / discourage use of  `\()` to define a function rather than `function()` but I wondered if the message could be changed, at least in cases where it is more clearly a function?  Regexp would look for (I think)

`= \(`
`, \(`
`<- \(`

as likely indicating a function rather than an escaped parenthesis.

**Comment by JWiley** [comment]
Oh I had tried to look at the NEWS but had not tried the development version. Yes it works great now.

**Comment by markschat** [comment]
Is it possible to rename this issue to contain "anonymous function" in the header? It was not obvious to me to find this topic here as I was facing the same problem.
E.g. "Anonymous functions (R 4.1.0) don´t pass linting"
